src/main/java/org/opensearch/flowframework/workflow/WorkflowProcessSorter.java

```diff
@@ -153,31 +153,45 @@ public void validate(List<ProcessNode> processNodes) throws Exception {
      * @throws Exception on validation failure
      */
     public void validatePluginsInstalled(List<ProcessNode> processNodes, WorkflowValidator validator) throws Exception {
-
-        // Retrieve node information to ascertain installed plugins
-        NodesInfoRequest nodesInfoRequest = new NodesInfoRequest();
-        nodesInfoRequest.clear().addMetric(NodesInfoRequest.Metric.PLUGINS.metricName());
-        CompletableFuture<List<String>> installedPluginsFuture = new CompletableFuture<>();
-        client.admin().cluster().nodesInfo(nodesInfoRequest, ActionListener.wrap(response -> {
-            List<String> installedPlugins = new ArrayList<>();
-
-            // Retrieve installed plugin names from the local node
-            String localNodeId = clusterService.state().getNodes().getLocalNodeId();
-            NodeInfo info = response.getNodesMap().get(localNodeId);
-            PluginsAndModules plugins = info.getInfo(PluginsAndModules.class);
-            for (PluginInfo pluginInfo : plugins.getPluginInfos()) {
-                installedPlugins.add(pluginInfo.getName());
-            }
-
-            installedPluginsFuture.complete(installedPlugins);
-
-        }, exception -> {
-            logger.error("Failed to retrieve installed plugins");
-            installedPluginsFuture.completeExceptionally(exception);
+        final CompletableFuture<List<String>> installedPluginsFuture = new CompletableFuture<>();
+
+        ClusterStateRequest clusterStateRequest = new ClusterStateRequest();
+        clusterStateRequest.clear().nodes(true).local(true);
```

Comment:
Do we need to add `clusterManagerNodeTimeout` for the clusterStateRequest as well? Or is it fine to ignore it since we are working with the local node?

Reply:
Correct, since we're restricting it to local we don't need the manager timeout.  And I've added our own timeout wrapping this whole wait, as a failsafe.